core/src/consensus.rs

```diff
@@ -377,24 +375,30 @@ impl Tower {
         }
     }
 
-    fn bank_weight(&self, bank: &Bank, ancestors: &HashMap<Slot, HashSet<Slot>>) -> u128 {
-        let (stake_lockouts, _) =
-            self.collect_vote_lockouts(bank.slot(), bank.vote_accounts().into_iter(), ancestors);
+    pub fn bank_lockouts(&self, bank: &Bank) -> (HashMap<Slot, StakeLockout>, u64) {
+        let mut ancestors = HashMap::new();
+        let mut history = vec![bank.slot()];
+        bank.slot_history()
+            .iter()
+            .take(MAX_ROOT_DEPTH)
```

Comment:
@carllin so I had to add this hyper parameter :(,   not sure if this has an impact on how big of a partition we can handle, or maybe it is fine.